### PR TITLE
feat: add reset-password UI

### PR DESCRIPTION
**Pull request by SwiftXLight**
**Description changes:**
- make UI for Reset Password form
- provide validation for new password
- check if Password and Confirm Password match

### General
- [ ] Assigned myself to the PR
- [ ] Assigned the appropriate labels to the PR
- [ ] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [ ] Performed a self-review of my code
- [ ] Types for input and output parameters
- [ ] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [ ] Don't have magic numbers
- [ ] Compare only with constants not with strings
- [ ] No ternary operator inside ternary operator
- [ ] Don't have commented code
- [ ] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [ ] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions is public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [ ] check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend
- [ ] Components and business logic are separated
- [ ] Colors, Font Size, and Font Name is on the theme or in the constants
- [ ] No text in the components, use i18n approach
- [ ] No inline styles
- [ ] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

**Overview:**
image
**Password don't match:**
image
**Passwords match:**
image

### CODE REVIEW

File: src/components/Auth/ResetPasswordForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    control,
+    formState: { errors, isValid }
+  } = useForm<ISignUp>({
+    mode: 'onChange',
+    defaultValues: {
+      password: '',
+      confirmPassword: ''
+    },
+    resolver: yupResolver(resetPasswordSchema),
+  });
```
[the remark this answers is not in the document]
Comment (lyudad) [reply src/components/Auth/ResetPasswordForm/index.tsx:45]: Screenshot 2023-04-05 at 18 07 13

File: src/components/Auth/ResetPasswordForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    </InputAdornment>
+                  </IconButton>
+                ),
+              }}
+            />
+          </AuthInput>
+          <AuthSendButton
+            disabled={!isValid}
+            type='submit'
+            value={t("Auth.save").toString()}
```
Comment (lyudad) [inline src/components/Auth/ResetPasswordForm/index.tsx]: t("Auth.save") is string

File: src/validation/auth.validate.ts [hunk 3]
```diff
@@ -34,3 +34,17 @@ export const signUpSchema = yup.object().shape({
...
     .oneOf([yup.ref('password')], '*Passwords does not match'),
 });
+
+export const resetPasswordSchema = yup.object().shape({
+  password: yup
+    .string()
+    .required('*Please Enter your password')
+    .matches(
+      PASSWORD_PATTERN,
+      '*Must Contain 10 Characters, One Uppercase, One Lowercase',
```
Comment (lyudad) [inline src/validation/auth.validate.ts]: 10 should be pass as parameter

Comment (SwiftXLight) [reply src/validation/auth.validate.ts]: fixed

File: src/components/Auth/ResetPasswordForm/index.tsx (hunk 2 above)
Comment (SwiftXLight) [reply src/components/Auth/ResetPasswordForm/index.tsx]: fixed

File: src/components/Auth/ResetPasswordForm/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (SwiftXLight) [reply src/components/Auth/ResetPasswordForm/index.tsx:45]: add blank line before resolver

**Review by lyudad — APPROVED** [review]

**Pull request merged**